### ISSUE TITLE
Andriod API 31+: Crash when receiving push notification

### ISSUE THREAD

**Issue by masterwok** [la-haus/iterable-flutter#40]
When targeting API 31 and above a crash occurs when receiving a push notification:

```
E/AndroidRuntime(30621): java.lang.IllegalArgumentException: com.coinme.coinme: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
E/AndroidRuntime(30621): Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
E/AndroidRuntime(30621): 	at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
E/AndroidRuntime(30621): 	at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:645)
E/AndroidRuntime(30621): 	at android.app.PendingIntent.getBroadcast(PendingIntent.java:632)
E/AndroidRuntime(30621): 	at com.iterable.iterableapi.IterableNotificationHelper$IterableNotificationHelperImpl.createNotification(IterableNotificationHelper.java:191)
E/AndroidRuntime(30621): 	at com.iterable.iterableapi.IterableNotificationHelper.createNotification(IterableNotificationHelper.java:36)
E/AndroidRuntime(30621): 	at com.iterable.iterableapi.IterableFirebaseMessagingService.handleMessageReceived(IterableFirebaseMessagingService.java:62)
E/AndroidRuntime(30621): 	at com.iterable.iterableapi.IterableFirebaseMessagingService.onMessageReceived(IterableFirebaseMessagingService.java:22)
E/AndroidRuntime(30621): 	at com.google.firebase.messaging.FirebaseMessagingService.dispatchMessage(FirebaseMessagingService.java:235)
E/AndroidRuntime(30621): 	at com.google.firebase.messaging.FirebaseMessagingService.passMessageIntentToSdk(FirebaseMessagingService.java:185)
E/AndroidRuntime(30621): 	at com.google.firebase.messaging.FirebaseMessagingService.handleMessageIntent(FirebaseMessagingService.java:172)
E/AndroidRuntime(30621): 	at com.google.firebase.messaging.FirebaseMessagingService.handleIntent(FirebaseMessagingService.java:161)
E/AndroidRuntime(30621): 	at com.google.firebase.messaging.EnhancedIntentService.lambda$processIntent$0$com-google-firebase-messaging-EnhancedIntentService(EnhancedIntentService.java:78)
E/AndroidRuntime(30621): 	at com.google.firebase.messaging.EnhancedIntentService$$ExternalSyntheticLambda1.run(Unknown Source:6)
E/AndroidRuntime(30621): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/AndroidRuntime(30621): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/AndroidRuntime(30621): 	at com.google.android.gms.common.util.concurrent.zza.run(com.google.android.gms:play-services-basement@@18.0.0:2)
E/AndroidRuntime(30621): 	at java.lang.Thread.run(Thread.java:920)
```

This looks like an upstream issue:

[https://github.com/Iterable/iterable-android-sdk/issues/430](https://github.com/Iterable/iterable-android-sdk/issues/430)

**Comment by danielgomezrico** [comment]
@masterwork thanks for reporting it and also link it to the native issue, we will check it

**Comment by cdmunoz** [comment]
@masterwok  as we have direct dependency with native SDK, whilst it's not tackled and fixed there before, it's impossible for us to have a Flutter fix.
Thank you very much for let us know about it

**Comment by masterwok** [comment]
@danielgomezrico @cdmunoz there was some activity on the upstream issue today. It looks like it could be as simple as updating the Iterable Android SDK to 3.4.2. I'll try bumping this and creating a PR sometime today or tomorrow.

**Comment by Santi92** [comment]
It's already available in `iterable_flutter: 0.5.4`
